src/docfx/lib/JsonUtility.cs

```diff
@@ -181,6 +207,51 @@ internal static List<Error> ValidateMismatchingFieldType(this JToken token, Type
             return errors;
         }
 
+        private static IList<Error> RegisterNewtonJsonSchemaLicense(string license)
+        {
+            if (string.IsNullOrEmpty(license))
+            {
+                return new List<Error>();
+            }
+
+            try
+            {
+                License.RegisterLicense(license);
+                return new List<Error>();
+            }
+            catch (Exception ex)
+            {
+                return new List<Error>() { Errors.NewtonJsonSchemaLicenseRegistrationFailed(ex.Message) };
+            }
+        }
+
+        private static (List<Error>, JSchema) GenerateJSchema(Type type)
+        {
+            try
+            {
+                if (_schemaGenerationEnabled != 1)
+                    return (new List<Error>(), null);
```

Comment:
also return limit exceeded error here?

Reply:
I am think about only logging the limit exceeded warning once, for instance if we generate 20 types during the run, it would be too many warnings in the log. In v2, we handled it the same way.

---
In reply to: [208167542](https://github.com/dotnet/docfx/pull/3164#discussion_r208167542) [](ancestors = 208167542)